pkg/client/dtclient/settings_client.go

```diff
@@ -616,7 +616,11 @@ func (d *SettingsClient) Upsert(ctx context.Context, obj SettingsObject, upsertO
 		return DynatraceEntity{}, err
 	}
 
-	log.WithCtxFields(ctx).Debug("Created/Updated object %s (%s) with externalId %s", obj.Coordinate.ConfigId, obj.SchemaId, externalID)
+	insertAfterForLogging := "<nil>"
+	if upsertOptions.InsertAfter != nil {
+		insertAfterForLogging = *upsertOptions.InsertAfter
```

Comment:
I think these lines would be better in a function

Reply:
9/10 times, I would also be on the side of a separate function, but here my gut feeling is that the if is so simple that i would make it more complicated. Since I would need to go into the function and then back up to the caller code. 
But if in the future this if needs to do more things then override the default value i would move it into a separate function.